pkg/executer/executer.go

```diff
@@ -0,0 +1,203 @@
+package executer
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/ocr2keepers/internal/util"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	pkgutil "github.com/smartcontractkit/ocr2keepers/pkg/util"
+)
+
+var ErrTooManyErrors = fmt.Errorf("too many errors in parallel worker process")
+
+type Registry interface {
+	CheckUpkeep(context.Context, bool, ...ocr2keepers.UpkeepKey) ([]ocr2keepers.UpkeepResult, error)
+}
+
+type Encoder interface {
+	// Eligible determines if an upkeep is eligible or not. This allows an
+	// upkeep result to be abstract and only the encoder is able and responsible
+	// for decoding it.
+	Eligible(ocr2keepers.UpkeepResult) (bool, error)
+	// Detail is a temporary value that provides upkeep key and gas to perform.
+	// A better approach might be needed here.
+	Detail(ocr2keepers.UpkeepResult) (ocr2keepers.UpkeepKey, uint32, error)
+	// SplitUpkeepKey ...
+	SplitUpkeepKey(ocr2keepers.UpkeepKey) (ocr2keepers.BlockKey, ocr2keepers.UpkeepIdentifier, error)
+}
+
+type Executer struct {
+	// injected dependencies
+	logger *log.Logger
+	reg    Registry
+	enc    Encoder
```

Comment:
🏄 I don't think we need to truncate the names of the components here?

Reply:
Probably not. My C style is getting out of hand. Lol